diff --git a/machineteaching/questions/get_dashboards.py b/machineteaching/questions/get_dashboards.py
index 7912991..e9a13f8 100644
--- a/machineteaching/questions/get_dashboards.py
+++ b/machineteaching/questions/get_dashboards.py
@@ -123,11 +123,11 @@ def get_on_time_exercises(user, chapters, onlineclass):
 
 
 def predict_drop_out(user, onlineclass, date):
-    # Get last completed chapter and model
-    completed_chapter = Deadline.objects.filter(onlineclass=onlineclass,
-                                                deadline__lte=date).order_by('deadline').last().chapter
-    model = completed_chapter.drop_out_model
     try:
+        # Get last completed chapter and model
+        completed_chapter = Deadline.objects.filter(onlineclass=onlineclass,
+                                                deadline__lte=date).order_by('deadline').last().chapter
+        model = completed_chapter.drop_out_model
         chapters = model.completed_chapter.all()
         # Open model file
         with open(model.model_file, "rb") as pklfile:
diff --git a/machineteaching/questions/views.py b/machineteaching/questions/views.py
index 0f75aa4..998bbbf 100644
--- a/machineteaching/questions/views.py
+++ b/machineteaching/questions/views.py
@@ -369,8 +369,8 @@ def show_outcome(request):
             problems = list(problems_all.values_list('id', flat=True))
             # Get latest student outcome for every student in class
             students = UserLogView.objects.filter(
-                user__userprofile__user_class=onlineclass,
-                problem_id__in=problems, timestamp__gte=onlineclass.start_date
+                user_class_id=onlineclass,
+                problem_id__in=problems
             ).order_by(Lower('user__first_name').asc(),
                        Lower('user__last_name').asc(),
                        'user_id',
@@ -897,4 +897,4 @@ def send_comment_email(student, comment, link):
                 <a href='{}' style='font-weight: 300; margin-left: 40%; width: 20%; height: 50%; padding: 0.5rem 1rem; justify-content: center; align-items: center; border-radius: 4px; text-decoration: none; color: white; background-color: #2196F3'>Veja aqui</a>            
             </div>
         </div>""".format(student.first_name, student.last_name, comment.userlog.problem.title, comment.user, comment.content, solution_link)
-    send_mail(message_subject, message_content, None, [student_email], fail_silently=False, html_message=message_html)
\ No newline at end of file
+    send_mail(message_subject, message_content, None, [student_email], fail_silently=False, html_message=message_html)
